### ISSUE TITLE
Would it be good to be able to use a Minecraft jar (as the one you control-click into in the IDE) that has mixins visibly applied?

### ISSUE THREAD

**Issue by natanfudge** [FabricMC/fabric-loom#165]
Place your votes

EDIT:
Just to be clear what this does:
Old:
```
void doMinecraftThing() {
    // Prints "Minecraft Fabric edition"
    System.out.println("Minecraft Java Edition");
}
```
New:
```
void doMinecraftThing() {
    // Prints "Minecraft Fabric edition"
   redirectorMixin();
}   

void redirectorMixin(){
     System.out.println("Minecraft Fabric Edition");
}

```
(A mixin is injected at this point that redirects the println to `println("Minecraft Fabric Edition")`)

**Comment by modmuss50** [comment]
I don’t think it’s really needed. Mixins aren’t super readable anyway so I don’t think it’s solves that issue.

Also makes it a right pita to disable mods in dev as you will have to regen the jar each time. Right now it’s easy to change something from to compile to provided and it will be gone from runtime without needing to reload gradle.

We tried it in the past, I don’t remember the details but I’m sure there is a good reason why it didn’t last long.

**Comment by natanfudge** [comment]
> I don’t think it’s really needed. Mixins aren’t super readable anyway so I don’t think it’s solves that issue.

Some unreadable code that amounts to what is actually happening is more readable than things happening with no indication of how or where they happen.

> Also makes it a right pita to disable mods in dev as you will have to regen the jar each time. Right now it’s easy to change something from to compile to provided and it will be gone from runtime without needing to reload gradle.

This could be a task you could run at will instead of running every time.

**Comment by natanfudge** [comment]
I'm interested if people have reasons other than what Modmuss mentioned.

**Comment by ghost** [comment]
Yes, If mixin source is still visible in other file and its clear where the mixin is and where it originates from.
e.g.
```java
// Begin net.fabricmc....WorldMixin.java
mixin code here
// End net.fabricmc....WorldMixin.java
```

**Comment by natanfudge** [comment]
Uhh, the mixin source will definitely stay where it is, but I don't know if the mixin'd minecraft jar will have these annotations.

**Comment by modmuss50** [comment]
One idea I came up with to incrase mixin visablity would be to add a comment to each class saying what mixins target the class.

Would look something like this:

```java
/**
 * Mixins:
 *     fabric-blockrenderlayer-v1: {@link net.fabricmc.fabric.mixin.blockrenderlayer.MixinBlockRenderLayer}
 */
class BlockRenderLayer {

}
```

This would require loom to look though the deps to find what mixins target each class, it might take a second or two, but I dont think it would be a major issue as it would only need to happen when using `genSources`.
